PMSCore/Admin.cs

```diff
@@ -0,0 +1,157 @@
+            // Placeholder until UserManager is implemented
+            return !string.IsNullOrWhiteSpace(userName) && userName == UserName;
+        }
+
+        public void SetPermission(Permission permission)
+        {
+            if (Enum.IsDefined(typeof(Permission), permission))
+            {
+                UserPermission = permission;
+            }
+        }
+
+        public Permission GetPermission() => UserPermission;
+    }
+
+    /// <summary>
+    /// The Admin user responsible for managing other users.
+    /// </summary>
+    public class Admin : UserBase
+    {
+        private readonly List<UserBase> _managedUsers;
+
+        /// <summary>
+        /// Initializes an Admin with a username.
+        /// </summary>
+        public Admin(string userName) : base(userName, Permission.ADMIN)
+        {
+            _managedUsers = new List<UserBase>();
+        }
+
+        /// <summary>
+        /// Creates a new user with a username and permission type.
+        /// </summary>
+        public UserBase? CreateUser(string userName, Permission permission)
+        {
+            if (string.IsNullOrWhiteSpace(userName))
+            {
+                _logError("Username cannot be empty.");
+                return null;
+            }
+
+            if (permission == Permission.DEFAULT || permission == Permission.ADMIN)
+            {
+                _logError("Invalid permission type for user creation.");
+                return null;
+            }
+
+            UserBase newUser = permission switch
+            {
+                Permission.TESTER => new Tester(userName),
+                Permission.DEVELOPER => new Developer(userName),
+                Permission.PROJECT_OWNER => new ProjectOwner(userName),
+                _ => null
+            };
+
+            if (newUser == null)
+            {
+                _logError("Failed to create user due to unsupported permission type.");
+                return null;
+            }
+
+            _managedUsers.Add(newUser);
+            _logInfo($"User {newUser.UserName} created with role {permission}.");
+            return newUser;
+        }
+
+        /// <summary>
+        /// Deletes a specified user from the list of managed users.
+        /// </summary>
+        public bool DeleteUser(UserBase user)
+        {
+            if (_managedUsers.Remove(user))
+            {
+                _logInfo($"User {user.UserName} has been removed.");
+                return true;
+            }
+
+            _logError("Specified user not found.");
+            return false;
+        }
+
+        /// <summary>
+        /// Returns a copy of the list of all users managed by the admin.
+        /// </summary>
+        public List<UserBase> ListUsers() => new List<UserBase>(_managedUsers);
+
+        /// <summary>
+        /// Updates an existing user's info.
+        /// </summary>
+        public bool UpdateUser(UserBase user, string newUserName, Permission newPermission)
```

Comment:
same as both comments above add the necessary methods if necessary in StateManager